src/AddressRegistry.Projections.Wms/AddressDetail/AddressDetailExtensions.cs

```diff
@@ -8,107 +8,50 @@ namespace AddressRegistry.Projections.Wms.AddressDetail
     using System.Threading.Tasks;
     using Microsoft.EntityFrameworkCore;
     using Microsoft.EntityFrameworkCore.Metadata.Internal;
+    using NetTopologySuite.Geometries;
+    using NetTopologySuite.IO;
 
     public static class AddressDetailExtensions
     {
+        private static string? STAsText(this Point point)
+            => point != null ? WKTWriter.ToPoint(point.Coordinate) : null;
+
         private static async Task<AddressDetailItem> FindAddressDetail(
             this WmsContext context,
             Guid addressId,
-            CancellationToken ct)
+            CancellationToken ct,
+            bool allowRemovedAddress = false)
         {
-            var address = await context
-                .Get<AddressDetailItem>()
-                .FindAsync(addressId, cancellationToken: ct);
+            // because of bulk inserts we needs to handle this a bit different
+            // NOTE: We cannot depend on SQL computed columns when facing with bulk insert that needs to perform queries.
+            // In case of an entry exists in the db but isn't available in local view. It also performs better than Load()
+            await context.AddressDetail.FindAsync(addressId, cancellationToken: ct);
+
+            // 'Local' in our case the source of truth for dealing with bulk operations
+            var address = context.AddressDetail.Local.Single(i => i.AddressId == addressId);
 
             if (address == null)
                 throw DatabaseItemNotFound(addressId);
 
-            return address;
+            // exclude soft deleted entries, unless allowed
+            if(!address.Removed || allowRemovedAddress)
+                return address;
+
+            throw DatabaseItemNotFound(addressId);
         }
 
         public static async Task<AddressDetailItem> FindAndUpdateAddressDetail(
             this WmsContext context,
             Guid addressId,
             Action<AddressDetailItem> updateFunc,
-            CancellationToken ct)
+            CancellationToken ct,
+            bool allowUpdateRemovedAddress = false)
```

Comment:
If `allowUpdateRemovedAddress` is always true, can we not remove it then?

Reply:
isn't it a bit illogical to update certain properties when it has been removed (in this case soft).
It's mostly to avoid those items from reoccurring in the results. unless explicitly requested.